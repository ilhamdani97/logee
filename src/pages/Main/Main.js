import React, { useEffect, useState } from 'react';
import styles from './styles.scoped.css';
import HeadingPort from '@components/elements/HeadingPort';
import { fetchData } from '../../actioncreator/action';
import { useDispatch, useSelector } from 'react-redux';
import { MainContext } from './context';
import { InfoBar } from 'logeeport-ui';
import CollapseInfo from '@components/elements/CollapseInfo';
import ContentGmv from '@components/elements/ContentGmv';
import { thousand } from '@utils/format';

export default function Main() {
  const[mainInit,setMainInit] = useState('Laporan Performa Bisnis');
  const[dataGmv, setDataGmv] = useState([]);
  const { data } = useSelector((s) => s.main);
  const dispatch = useDispatch();

  useEffect(() => {
    const fetchingData = data.fetchDashboardGmv && data.fetchDashboardGmv.data;
    const dataFetchGmv = [
      {
        title: 'GVM',
        subTitle: 'Akumulasi seluruh transaksi yang berhasi dilakukan dalam periode waktu yang dipilih',
        total: `Rp ${thousand(fetchingData? fetchingData.gmv.total : 0)}`,
        collaps: true,
        content: [
          {
            title: 'Transaksi Kontainer',
            total: `Rp ${thousand(fetchingData? fetchingData.gmv.container : 0)}`
          },
          {
            title: 'Transaksi Pesanan Truk',
            total: `Rp ${thousand(fetchingData? fetchingData.gmv.truckOrder : 0)}`
          }
        ]
      },
      {
        title: 'Transaksi',
        subTitle: 'Semua transaksi dari semua metode pembayaran periode waktu yang dipilih.',
        total: `${thousand(fetchingData? fetchingData.transaction.total : 0)}`,
        collaps: false,
        content: []
      },
      {
        title: 'Revenue',
        subTitle: 'Total pendapatan dari seluruh transaksi pada periode waktu yang dipilih',
        total: `Rp ${thousand(fetchingData? fetchingData.revenue.total : 0)}`,
        collaps: true,
        content: [
          {
            title: 'Kontainer Paylater',
            total: `Rp ${thousand(fetchingData ? fetchingData.revenue.containerPayLater : 0)}`
          },
          {
            title: 'Pesanan Truk Paylater',
            total: `Rp ${thousand(fetchingData? fetchingData.revenue.orderTruckPayLater : 0)}`
          },
          {
            title: 'Pesanan Truk Invoice Acceptence',
            total: `Rp ${thousand(fetchingData?
              fetchingData.revenue.orderTruckInvoiceAcceptance : 0)}`
          },
          {
            title: 'Charge fee Kontainer',
            total: `Rp ${thousand(fetchingData?
              fetchingData.revenue.chargeFeeContainer : 0)}`
          },
          {
            title: 'Charge fee Pesanan Truk',
            total: `Rp ${thousand(fetchingData?
              fetchingData.revenue.chargeFeeOrderTruck : 0)}`
          }
        ]
      }
    ];
    dispatch(fetchData('fetchUserList', '', 'limit=10', 'MainApi'));
    dispatch(fetchData('fetchDashboardGmv', '', '', 'MainApi'));
    setDataGmv(dataFetchGmv);
  }, []);
  return (
    <MainContext.Provider value={{
      mainInit,setMainInit
    }}>
      <div className={styles.root}>
        <HeadingPort title={mainInit} />
        <InfoBar
          color="primary"
          text="Periode laporan: 2022"
        />
        {dataGmv && dataGmv.map((item, i) =>
          (<CollapseInfo
            children = {
              <>
                {item.content && item.content.map((item, i) =>
                  <ContentGmv data={item} key={i}/>
                )}
              </>
            }
            collaps = {item.collaps}
            header={item}
            key={i}
          />)
        )}
      </div>
    </MainContext.Provider>
  );
}
