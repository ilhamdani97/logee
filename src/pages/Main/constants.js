import React from 'react';
export const SUCCESS = 'success';
export const RESET = 'reset';
export const DATA_FETCHED = 'Dashboard/data-fetched';
export const LOADING = 'Dashboard/loading';
export const FAILED = 'Dashboard/failed';
export const CLEAR_MESSAGE = 'Dashboard/clearmessage';

export const dataCaption = [{
  title:'Transaksi & Kontainer',
  type:'fetchChartProforma',
  xLabel:'periodNode',
  yLabel:'totalTransaction',
  tooltip:{
    titleUpper:{
      currency:false,
      value:'totalContainer',
      subCaption:{
        textPosition:'right',
        textValue:'Kontainer'
      },
      tooltipColor:'primary'
    },
    titleBottom:{
      currency:true,
      value:'totalTransaction',
      subCaption:{
        textPosition:'right',
        textValue:''
      },
      tooltipColor:'primary'
    }
  },
  averageInfo:<p style={{ textAlign:'center' }}>Rata-rata jumlah kontainer dan transaksinya <br />
  dalam satu periode waktu yang dipilih.</p>,
  averageCaption:{
    titleLeft:'Kontainer',
    titleRight:'Transaksi',
    valueLeft:{
      currency:false,
      value:'averageContainer'
    },
    valueRight:{
      currency:true,
      value:'averageTransaction'
    }
  },
  barType:'normal',
},
{
  title:'Ekspor & Impor',
  type:'fetchExportImport',
  xLabel:'periodNode',
  yLabel:'totalTransactionImport',
  tooltip:{
    titleUpper:{
      currency:true,
      value:'totalTransactionImport',
      subCaption:{
        textPosition:'left',
        textValue:'Impor : '
      },
      tooltipColor:'primary'
    },
    titleBottom:{
      currency:true,
      value:'totalTransactionExport',
      subCaption:{
        textPosition:'left',
        textValue:'Ekpor : '
      },
      tooltipColor:'secondary'
    }
  },
  averageInfo:<p style={{ textAlign:'center' }}>Rata-rata jumlah transaksi ekspor dan impor <br />
  dalam satu periode waktu yang dipilih.</p>,
  averageCaption:{
    titleLeft:'Impor',
    titleRight:'Ekspor',
    valueLeft:{
      currency:true,
      value:'averageTransactionImport'
    },
    valueRight:{
      currency:true,
      value:'averageTransactionExport'

    }
  },
  barType:'stack',
},
{
  title:'Pesanan Truk',
  type:'fetchPesananTruck',
  xLabel:'periodNode',
  yLabel:'totalTransaction',
  tooltip:{
    titleUpper:{
      currency:false,
      value:'totalOrderCount',
      subCaption:{
        textPosition:'right',
        textValue:'Pesanan'
      },
      tooltipColor:'primary'
    },
    titleBottom:{
      currency:true,
      value:'totalTransaction',
      subCaption:{
        textPosition:'right',
        textValue:''
      },
      tooltipColor:'primary'
    }
  },
  averageInfo:<p style={{ textAlign:'center' }}>Rata-rata jumlah pesanan dan transaksinya <br />
  dalam satu periode waktu yang dipilih.</p>,
  averageCaption:{
    titleLeft:'Pesanan Truk',
    titleRight:'Transaksi Truk',
    valueLeft:{
      currency:false,
      value:'averageOrderCount'
    },
    valueRight:{
      currency:true,
      value:'averageTransaction'
    }
  },
  barType:'normal',
}];

export const optionsArray = [
  { text: 'NPCT1', value: 'NPCT1' },
  { text: 'KOJA', value: 'KOJA' },
  // { text: 'NLE', value: 'NLE' },
  // { text: 'JICT', value: 'JICT' },
];

export const tab = [
  { 'title': 'Semua', 'active': true, 'status': 'false', alias:'ALL', totalName:'totalALL' },
  { 'title': 'Memproses Proforma', 'active': false, 'status': 'true', alias:'CREATED',totalName:'totalCREATED' },
  { 'title': 'Menunggu Pembayaran', 'active': false, 'status': 'true', alias:'PROCESSED',totalName:'totalPROCESSED' },
  { 'title': 'Sudah dibayar', 'active': false, 'status': 'true', alias:'PAID',totalName:'totalPAID' },
];


export const proformaToolTip = {
  'paid': <p>Total transaksi proforma pengurusan baru <br />dan perpanjangan yang telah dibayar PPJK</p>,
  'unpaid': <p>Total transaksi proforma pengurusan baru <br />dan perpanjangan yang belum dibayar PPJK</p>
};



export const akusisiToolTip = {
  'aktif': {
    'title': 'Akun Aktif',
    'tooltip': <p>Total akun PPJK yang <br />berhasil diaktivasi oleh BisOps.</p>
  },
  'nonAktif': {
    'title': 'Akun Belum Aktif',
    'tooltip': <p>Total akun PPJK yang <br />belum diaktivasi oleh BisOps.</p>
  },
  'akunKonek': {
    'title': 'User Terhubung',
    'tooltip': <p>Total user yang terhubung <br />dengan PPJK</p>
  },
  'trukJoin': {
    'title': 'Truk Bergabung',
    'tooltip': <p>Jumlah truk yang baru ditambahkan <br />mitra Trucker untuk kebutuhan PPJK</p>
  },
};
