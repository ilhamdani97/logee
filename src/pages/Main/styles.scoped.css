.root {
  padding: 1.6rem;
  margin: 8px;
  background-color: #ffffff;
  border-radius: 16px;
  margin-top: 1.6rem;
}

.root > h1 {
  left: 50%;
  margin: auto;
  position: absolute;
  text-align: center;
  top: 50%;
  transform: translateX(-50%) translateY(-50%);
}

.root > div:nth-child(2) {
  margin-top: 30px;
}

.dashboardCollaps {
  background-color: #ffffff;
  margin: 0 12px;
}

.periodeRight {
  text-align: left;
}

.hidden {
  display: none;
}
@media only screen and (max-width: 48em) {
  .root {
    padding: 1rem;
  }
}
