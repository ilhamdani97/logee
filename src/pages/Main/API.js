import { fetchUserList } from '../../services/user';
import { fetchDashboardGmv } from '@services/dashboard';

import { DATA_FETCHED, LOADING, FAILED } from './constants';

export const API = {
  fetchUserList,
  fetchDashboardGmv
};

export const TYPES = {
  DATA_FETCHED, LOADING, FAILED
};

