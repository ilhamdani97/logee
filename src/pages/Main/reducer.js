import { DATA_FETCHED, LOADING, FAILED, CLEAR_MESSAGE } from './constants';

const initialState = {
  isLoading: {
    fetchUserList:false
  },
  data:{
    fetchUserList:false
  },
  message:{
    fetchUserList:false
  },
};

export default function reducer(state = initialState, action = {}) {
  const { type, data, isLoading, key, message  } = action;

  switch (type) {
    case DATA_FETCHED:
      return {
        ...state,
        isLoading: { ...state.isLoading, [key] : isLoading },
        data:{ ...state.data, [key]: data }
      };
    case LOADING:
      return {
        ...state,
        isLoading: { ...state.isLoading, [key] : isLoading },
      };
    case FAILED:
      return {
        ...state,
        isLoading: { ...state.isLoading, [key] : isLoading },
        message: { ...state.message, [key] : message },
      };
    case CLEAR_MESSAGE:
      return {
        ...state,
        message: { ...state.message, [key]:message },
      };

    default:
      return state;
  }
}
