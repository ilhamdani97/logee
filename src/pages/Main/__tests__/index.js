import React from 'react';
import { useSelector } from 'react-redux';
import ShallowRenderer from 'react-test-renderer/shallow';
import Main from '../Main';

describe('src/pages/Main', () => {
  test('render', () => {
    useSelector.mockImplementation((fn) => {
      fn({});
      return {
        data:{
          'fetchChartProforma': {
            'success': true,
            'data': {
              'period': 'yearly',
              'year': 2021,
              'month': '-',
              'week': '-',
              'date': '-',
              'averageContainer': 3.9166666666666665,
              'averageTransaction': 2812512994.1666665,
              'chart': [
                {
                  'periodNodeId': 1,
                  'periodNode': 'Jan',
                  'totalTransaction': 967522920,
                  'totalContainer': 1
                },

              ]
            },
            'message': 'Your Request Has Been Processed',
            'code': 200,
            'statusCode': '2000'
          },
          'fetchExportImport': {
            'success': true,
            'data': {
              'period': 'yearly',
              'year': 2021,
              'month': '-',
              'week': '-',
              'date': '-',
              'averageTransactionExport': 1706852.5,
              'averageTransactionImport': 2810806141.6666665,
              'chart': [
                {
                  'totalTransactionExport': 0,
                  'totalTransactionImport': 967522920,
                  'periodNodeId': 1,
                  'periodNode': 'Jan',
                  'totalTransactionAll': 967522920
                },
              ]
            },
            'message': 'Your Request Has Been Processed',
            'code': 200,
            'statusCode': '2000'
          },
          'fetchPesananTruck': {
            'success': true,
            'data': {
              'period': 'yearly',
              'year': 2021,
              'month': '-',
              'week': '-',
              'date': '-',
              'averageOrderCount': 5.416666666666667,
              'averageTransaction': 9194735.916666666,
              'chart': [
                {
                  'periodNodeId': 1,
                  'periodNode': 'Jan',
                  'totalTransaction': 7500000,
                  'totalOrderCount': 3
                },

              ]
            },
            'message': 'Your Request Has Been Processed',
            'code': 200,
            'statusCode': '2000'
          },
          'fetchDashboardOrders': {
            'success': true,
            'data': {
              'from': '2021-12-26T17:00:00.000Z',
              'to': '2021-12-27T16:59:59.999Z',
              'latestUpdate': '2021-12-27T08:03:32.603Z',
              'totalOnProcess': 0,
              'totalOnConfirm': 0,
              'totalOnDelivery': 0,
              'totalDone': 0,
              'totalExpired': 0,
              'totalOrders': 0,
              'percentageOnProcess': 0,
              'percentageOnConfirm': 0,
              'percentageOnDelivery': 0,
              'percentageDone': 0,
              'percentageExpired': 0
            },
            'message': 'Your Request Has Been Processed',
            'code': 200,
            'statusCode': '2000'
          },
          'fetchDashboardProforma': {
            'success': true,
            'data': {
              'from': '2021-12-26T17:00:00.000Z',
              'to': '2021-12-27T16:59:59.999Z',
              'latestUpdate': '2021-12-27T08:03:32.545Z',
              'totalPaid': 0,
              'totalUnpaid': 0
            },
            'message': 'Your Request Has Been Processed',
            'code': 200,
            'statusCode': '2000'
          },
          'fetchDashboardGmv': null
        }
      };
    });
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<Main />);
    expect(tree).toMatchSnapshot();
  });
});
