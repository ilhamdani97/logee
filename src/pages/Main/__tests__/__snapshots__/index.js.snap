// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/pages/Main render 1`] = `
<Provider
  value={
    Object {
      "mainInit": "dashboard page",
      "setMainInit": [MockFunction],
    }
  }
>
  <div>
    <HeadingPort
      title="dashboard page"
    />
    <Typography
      spacing=""
      variant="h1"
    >
      Selamat datang di Logee
    </Typography>
  </div>
</Provider>
`;
